src/mqtt_client/mqtt_client.rs

```diff
@@ -115,7 +203,8 @@ where
             self.subscribed = true;
```

Comment:
This is now big enough so that it should not be a collection of flags (`subscribed`, `Option<Timeout>`, `Option<Iterator>`, `iteration_exhausted`) but a state machine. Let's do this comprehensively.

Reply:
Yeah, I was noticing this as well. Will refactor